packages/itwinui-react/src/react-table/react-table-types.test.tsx

```diff
@@ -0,0 +1,802 @@
+        },
+        disableResizing: true,
+      },
+      {
+        ...ActionColumn({ columnManager: true }),
+        Cell: (props: TableTypes.CellProps<TableRowDataType>) => (
+          <DropdownMenu menuItems={menuItems}>
+            <IconButton
+              styleType='borderless'
+              onClick={(e) => e.stopPropagation()}
+              disabled={isRowDisabled(props.row.original)}
+            >
+              <SvgMore />
+            </IconButton>
+          </DropdownMenu>
+        ),
+        sticky: 'right',
+      },
+    ],
+    [
+      isCheckboxDisabled,
+      expandedSubComponent,
+      isExpanderDisabled,
+      translatedLabels,
+      cellRenderer,
+      formatDate,
+      menuItems,
+      isRowDisabled,
+    ],
+  ) satisfies TableTypes.Column<TableRowDataType>[];
+
+  const data = React.useMemo(
+    () => [
+      {
+        index: 1,
+        name: 'Name1',
+        description: 'Description1',
+        ids: ['1'],
+        startDate: new Date('May 1, 2021'),
+        endDate: '2021-05-31T21:00:00.000Z',
+        subRows: [
+          {
+            index: 11,
+            name: 'Name11',
+            description: 'Description11',
+            ids: ['11'],
+            startDate: new Date('May 11, 2021'),
+            endDate: '2021-05-31T21:00:00.000Z',
+            subRows: [
+              {
+                index: 111,
+                name: 'Name111',
+                description: 'Description111',
+                ids: ['111'],
+                startDate: new Date('May 11, 2021'),
+                endDate: '2021-05-31T21:00:00.000Z',
+                subRows: [],
+              },
+            ],
+          },
+        ],
+      },
+      {
+        index: 2,
+        name: 'Name2',
+        description: 'Description2',
+        ids: ['2', '3', '4'],
+        startDate: new Date('May 2, 2021'),
+        endDate: '2021-06-01T21:00:00.000Z',
+        subRows: [
+          {
+            index: 21,
+            name: 'Name21',
+            description: 'Description21',
+            ids: ['21'],
+            startDate: new Date('May 21, 2021'),
+            endDate: '2021-06-01T21:00:00.000Z',
+            subRows: [],
+          },
+          {
+            index: 22,
+            name: 'Name22',
+            description: 'Description22',
+            ids: ['22'],
+            startDate: new Date('May 22, 2021'),
+            endDate: '2021-06-01T21:00:00.000Z',
+            subRows: [],
+          },
+        ],
+      },
+      {
+        index: 3,
+        name: 'Name3',
+        description: 'Description3',
+        ids: ['3', '4'],
+        startDate: new Date('May 3, 2021'),
+        endDate: '2021-06-02T21:00:00.000Z',
+        subRows: [],
+      },
+    ],
+    [],
+  ) satisfies TableRowDataType[];
+
+  const onFilter = React.useCallback(
+    (
+      filters: TableFilterValue<TableRowDataType>[],
+      state: TableTypes.TableState,
+      filteredData: TableTypes.Row<TableRowDataType>[],
```

Comment:
i think these parameters don't need to be explicitly typed since the whole function is already being typed

Reply:
https://github.com/iTwin/iTwinUI/pull/1514/commits/2f8cec68743d317548640d40123aadba4d25654d: Sounds good, removed